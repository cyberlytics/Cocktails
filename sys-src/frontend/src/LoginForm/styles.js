export const styles = {
    styleDiv : {
        position: "center",
        height: "310px",
        width: "400px",
        top: "50px",
        left: "50%",

        backgroundColor: 'rgba(0, 0, 0, 60%)',
        boxSizing: 'border-box',
    },
    styleInputField : {
        margin: '10px 10px',
        overflow: 'auto',
    },    
    styleButton : {
        backgroundColor: 'rgb(207, 181, 59)',
        marginTop: '5px',
        width: '99%',
        height: '50px',
        fontSize: '22px',
        padding: '0',
        border: 'none',
    },
    styleHeadline : {
        color: 'white',
        textAlign: 'center',
        paddingTop : '10px',
    },
    styleLinkText : {
        color: 'white',
        textAlign : 'center',
        fontSize: '21px',
    }
};