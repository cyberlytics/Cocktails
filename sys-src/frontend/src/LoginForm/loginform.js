//Import modules
import React, { Component } from 'react';

//Import own UI-Elements
import IconTextField from '../inputs/icontextfield';

//Import local ressources
import { styles } from './styles';
import { apiurl } from '../api.js';

class LoginForm extends Component {
    
    //Initialize username and password field with empty string and enable button
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            buttonDisabled: false
        };
        this.onKeyUp = this.handleKeypress.bind(this);
    }

    resetForm() {
        this.setState({
            username: '',
            password: '',
            buttonDisabled: false
        })
    }

    componentDidMount() {
        this.resetForm();
    }

    setInputValue(property, val) {
        this.setState({
            [property] : val
        })
    }

    async handleSubmit() {
        if (!this.state.username) {
            return;
        }
        if (!this.state.password) {
            return;
        }

        this.setState({
            buttonDisabled : true
        })

        try {
            let res = await fetch(apiurl + '/login', {
                method: "post",
                headers: {
                    'Accept': 'application/json',
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                    username: this.state.username, 
                    password: this.state.password
                })
            });
            
            //process result
            let result = await res.json();

            //Successful authenticated
            if (result && result.success) {
                localStorage.setItem('isLoggedInId', result.id);
                window.location.href='/';
                return true;
            }
            else if (result && result.success === false) {
                alert(result.msg);
                this.resetForm();
                window.location.href='/login';
                return false;
            }
        } catch (error) {
            alert("Fehler:" + error.message);
            this.resetForm();
            return false;
        }
    }

    handleKeypress = e => {
        if (e.key === "Enter") {
          this.handleSubmit();  
        }
    };

    render() { 
        return (
            <div style={styles.styleDiv} className="login-form" onKeyPress={this.handleKeypress}>
                <h1 style={styles.styleHeadline}>Login</h1>
                <div style={styles.styleInputField}><IconTextField type="username" placeholder="Username" iconclass="fas fa-user"
                value={this.state.username ? this.state.username : ''} onChange={ (val) => this.setInputValue('username', val)} /></div>
                <div style={styles.styleInputField}><IconTextField type="password" placeholder="Password" iconclass="fas fa-lock"
                value={this.state.password ? this.state.password : ''} onChange={ (val) => this.setInputValue('password', val)} /></div>
                <div style={styles.styleInputField}><button style={styles.styleButton} variant="contained" type="submit" className="btn btn-primary"
                onClick={() => this.handleSubmit() } disabled={this.state.buttonDisabled}>Login</button></div>
                <div style={styles.styleLinkText}><span>Not registered yet? Click <a href="/register">Here</a></span></div>
            </div>
        );
    }
}
 
export default LoginForm;