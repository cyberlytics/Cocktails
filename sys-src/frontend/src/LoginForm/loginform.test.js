import React from 'react';
import { render } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import { expect, it } from '@jest/globals';
import TestUtils from 'react-dom/test-utils'

//Import own modules
import LoginForm from './loginform';


describe('Tests for LoginForm', () => {
    test("Test headline in LoginForm rendering", () => {
        const { getAllByText } = render(<LoginForm/>);
        expect(getAllByText(/register/i)[0]).toBeInTheDocument();
    });

    test("renders correctly Username", () => {
        const {queryByPlaceholderText} = render(<LoginForm/>);
        const input = queryByPlaceholderText("Username");
        expect(input).toBeTruthy();
    });

    test("renders correctly Password", () => {
        const {queryByPlaceholderText} = render(<LoginForm/>);
        const input = queryByPlaceholderText("Password");
        expect(input).toBeTruthy();
    });

    test("all IconTextFields rendered correctly", () => {
        const { getAllByTestId } = render(<LoginForm/>);
        const nodes = getAllByTestId('icon-textfield');
        expect(nodes.length).toBe(2);
    });

    test("renders correctly Submit Button", () => {
        const { queryAllByText } = render(<LoginForm/>);
        const button = queryAllByText("Login")[1];
        expect(button).toBeTruthy();
    });

    test("renders correctly Register Link", () => {
        const { queryByText } = render(<LoginForm/>);
        const button = queryByText("Here");
        expect(button).toBeTruthy();
    });

    it('input of username field works', () => {
        const textComponent = TestUtils.renderIntoDocument(
            <LoginForm/>
        );
        const node = TestUtils.scryRenderedDOMComponentsWithTag(textComponent, 'input')[0];
        TestUtils.Simulate.change(node, { target: { value: 'john'}});
        expect(textComponent.state.username).toEqual("john");
    });
    
    it('input of password field works', () => {
        const textComponent = TestUtils.renderIntoDocument(
            <LoginForm/>
        );
        const node = TestUtils.scryRenderedDOMComponentsWithTag(textComponent, 'input')[1];
        TestUtils.Simulate.change(node, { target: { value: '12345'}});
        expect(textComponent.state.password).toEqual("12345");
    });

    it('submit button works', () => {
        const component = TestUtils.renderIntoDocument(
            <LoginForm/>
        );
        const node = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
        component.state.username = "john";
        component.state.password = "12345";

        TestUtils.Simulate.click(node);

        //Form resetted
        expect(true).toBe(true);
    })

    it('submit with Enter key press works', () => {
        const component = TestUtils.renderIntoDocument(
            <LoginForm/>
        );
        const node = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
        component.state.username = "john";
        component.state.password = "12345";

        TestUtils.Simulate.keyPress(node, { key : 'Enter'});

        //see if nothing has changed
        expect(component.container).toMatchSnapshot();
    })

    it('no submit with empty username', () => {
        const component = TestUtils.renderIntoDocument(
            <LoginForm/>
        );
        const node = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
        component.state.username = "";
        component.state.password = "12345";

        TestUtils.Simulate.click(node);

        //see if nothing has changed
        expect(component.container).toMatchSnapshot();
    });

    it('no submit with empty password', () => {
        const component = TestUtils.renderIntoDocument(
            <LoginForm/>
        );
        const node = TestUtils.findRenderedDOMComponentWithTag(component, 'button');
        component.state.username = "john";
        component.state.password = "";

        TestUtils.Simulate.click(node);

        //see if nothing has changed
        expect(component.container).toMatchSnapshot();
    });

    //Unfinished
    // it('test handleSubmit call', async () => {
    //     const component = TestUtils.renderIntoDocument(
    //         <LoginForm/>
    //     );
    //     component.state.username = "john";
    //     component.state.password = "12345";
    //     let success = await component.handleSubmit();
    //     expect(success).toBeFalsy();
    // });
})
