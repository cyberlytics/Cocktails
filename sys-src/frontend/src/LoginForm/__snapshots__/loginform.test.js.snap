// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for LoginForm no submit with empty password 1`] = `undefined`;

exports[`Tests for LoginForm no submit with empty username 1`] = `undefined`;

exports[`Tests for LoginForm submit with Enter key press works 1`] = `undefined`;
